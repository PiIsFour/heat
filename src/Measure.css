.Measure {
	border: 2px solid gray;
	padding: .5em;
	margin: .5em;
	text-align: center;
}

.Measure input {
	text-align: right;
	margin-left: .5em;
	margin-right: .5em;
	max-width: 5em;
}

.Measure > div {
	margin: .5em;
}
