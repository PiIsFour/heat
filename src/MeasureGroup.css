.MeasureGroup {
	display: flex;
	flex-direction: column;
	justify-content: space-evenly;
	align-items: center;
	border: 2px solid gray;
	padding: .5em;
	/* margin: 1em; */
}

.MeasureGroup > .name {
	margin: .5em;
	font-size: 1.3em;
}
