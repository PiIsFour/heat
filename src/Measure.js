import React, {Component} from 'react'
import './Measure.css'

class Measure extends Component {
	constructor (props) {
		super(props)
		this.handleValueUpdate = this.handleValueUpdate.bind(this)
		this.handleInputUpdate = this.handleInputUpdate.bind(this)
	}
	handleValueUpdate (event) {
		const {valueChanged, name} = this.props.state
		valueChanged(name, event.target.value)
	}
	handleInputUpdate (event) {
		const {inputToggled, name, input} = this.props.state
		inputToggled(name, !input)
	}
	render () {
		const {unit, input, calculated, symbol, value, longname, formul} = this.props.state
		const style = {}
		if (!input && !calculated) {
			style.backgroundColor = 'gray'
		}
		return <div className="Measure" style={style}>
			<div>{longname}</div>
			<div>
				<input className="input" type="checkbox" checked={input} disabled={calculated} onChange={this.handleInputUpdate}/>
				<math xmlns="http://www.w3.org/1998/Math/MathML">{symbol}</math>
				<input className="value" type="text" readOnly={!input} disabled={!input && !calculated} value={calculated ? value.toFixed(2) : value} onChange={this.handleValueUpdate}/>
				<span className="unit">{unit}</span>
			</div>
			<div>{formul}</div>
		</div>
	}
}

export default Measure
