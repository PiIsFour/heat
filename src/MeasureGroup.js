import React, {Component} from 'react'
import './MeasureGroup.css'

import Measure from './Measure'

class MeasureGroup extends Component {
	render () {
		const {state, opt} = this.props
		return <div className="MeasureGroup" style={opt.style}><div className="name">{opt.name}</div>
			{
				opt.childs
					.map(n => ({...state[n], ...state.all, name: n, inactive: opt.inactive}))
					.map(state => (<Measure key={state.name} state={state}/>))
			}
		</div>
	}
}

export default MeasureGroup
