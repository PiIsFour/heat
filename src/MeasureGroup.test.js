import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import MeasureGroup from './MeasureGroup'
import Measure from './Measure'
import composeState from './state'

describe('MeasureGroup', () => {
	const state = composeState().compose('twin', 'kg', 42, false, 'Temperatur', <msub><mi>T</mi><mi>W in</mi></msub>)
	state.all = {
		valueChanged: jest.fn(),
		inputToggled: jest.fn()
	}
	const opt = {
		name: 'testname',
		childs: ['twin']
	}
	const component = shallow(
		<MeasureGroup state={state} opt={opt}/>
	)

	it('renders and matches our snapshot', () => {
		const component = renderer.create(
			<MeasureGroup state={state} opt={opt}/>
		)
		const tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('contain same nuber of childs as given', () => {
		expect(component.find(Measure)).toHaveLength(opt.childs.length)
	})
})
