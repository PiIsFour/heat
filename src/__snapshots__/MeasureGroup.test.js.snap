// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeasureGroup renders and matches our snapshot 1`] = `
<div
  className="MeasureGroup"
  style={undefined}
>
  <div
    className="name"
  >
    testname
  </div>
  <div
    className="Measure"
    style={Object {}}
  >
    <div />
    <div>
      <input
        checked={
          <mrow>
            <msub>
              <mi>
                T
              </mi>
              <mi>
                W in
              </mi>
            </msub>
          </mrow>
        }
        className="input"
        disabled={false}
        onChange={[Function]}
        type="checkbox"
      />
      <math
        xmlns="http://www.w3.org/1998/Math/MathML"
      />
      <input
        className="value"
        disabled={false}
        onChange={[Function]}
        readOnly={false}
        type="text"
        value={42}
      />
      <span
        className="unit"
      >
        kg
      </span>
    </div>
    <div>
      
    </div>
  </div>
</div>
`;
